go-controller/hybrid-overlay/pkg/controller/node_linux.go

```diff
@@ -54,26 +56,69 @@ func podToCookie(pod *kapi.Pod) string {
 	return nameToCookie(pod.Namespace + "_" + pod.Name)
 }
 
-func (n *NodeController) addOrUpdatePod(pod *kapi.Pod) error {
+func (n *NodeController) addOrUpdatePod(pod *kapi.Pod, wf *factory.WatchFactory) error {
 	podIPs, podMAC, err := getPodDetails(pod, n.nodeName)
 	if err != nil {
 		klog.V(5).Infof("cleaning up hybrid overlay pod %s/%s because %v", pod.Namespace, pod.Name, err)
 		return n.deletePod(pod)
 	}
 
+	namespace, err := ovn.WaitForNamespace(pod.Namespace, wf)
+	if err != nil {
+		return err
+	}
+	namespaceExternalGW := namespace.GetAnnotations()[hotypes.HybridOverlayExternalGw]
 	cookie := podToCookie(pod)
 	for _, podIP := range podIPs {
 		_, _, err = util.RunOVSOfctl("add-flow", extBridgeName,
 			fmt.Sprintf("table=10,cookie=0x%s,priority=100,ip,nw_dst=%s,actions=set_field:%s->eth_src,set_field:%s->eth_dst,output:ext", cookie, podIP.IP, n.drMAC, podMAC))
 		if err != nil {
 			return fmt.Errorf("failed to add flows for pod %s/%s: %v", pod.Namespace, pod.Name, err)
 		}
+		if namespaceExternalGW != "" {
+			_, _, err = util.RunOVSOfctl("add-flow", extBridgeName,
+				fmt.Sprintf("table=0, cookie=0x%s, priority=100, ip, nw_src=%s, actions=load:%s->NXM_NX_TUN_ID[0..31],set_field:%s->tun_dst,output:%s", cookie, podIP.IP, strconv.Itoa(hotypes.HybridOverlayVNI), namespaceExternalGW, extVXLANName))
+			if err != nil {
+				return fmt.Errorf("failed to add flows for pod %s/%s: %v", pod.Namespace, pod.Name, err)
+			}
+			node, err := wf.GetNode(n.nodeName)
```

Comment:
@rcarrillocruz n.nodeName is the local node, and we already have its MAC in n.drMAC so we don't need to look the node up.

Reply:
ok, let's use drMAC.
OTOH, we do need to pass the node subnet to the flow, I don't see how I can pull that value without getting the Node object?

Reply:
ok, as agreed on Slack, we don't care whether it's drMAC or br-ext, I'll change to drMAC then to avoid calls

Reply:
@rcarrillocruz yeah, as long as the packets go through, we don't care :)